KIPs/kip-37.md

```diff
@@ -0,0 +1,908 @@
+
+The optional `KIP37Mintable` extension can be identified with the (KIP-13 Standard Interface Detection)[https://kips.klaytn.com/KIPs/kip-13].
+
+If the optional `KIP37Mintable` extension is included:
+
+- The KIP-13 `supportsInterface` function MUST return the constant value `true` if `0x84aec3b9` is passed through the `interfaceID` argument.
+- The `create` function is used to create new token allocated with new token id.
+  - An implementation MUST emit the `URI` event during a create operation if the created token has it's own metadata.
+- When minting/creating tokens, the `_from` argument MUST be set to `0x0` (i.e. zero address). See [Minting-creating and burning-destroying rules](#minting-creating-and-burning-destroying-rules).
+
+```solidity
+pragma solidity 0.5.6;
+
+/// @title KIP-37 Multi Token Standard, optional minting extension
+///  Note: the KIP-13 identifier for this interface is 0x84aec3b9.
+interface IKIP37Mintable {
+    /// @notice Creates a new token type and assings _initialSupply to minter
+    /// @dev Throws if `msg.sender` is not allowed to create
+    ///   Throws if the token id is already used.
+    /// @param _id The token id to create.
+    /// @param _initialSupply The amount of tokens being minted.
+    /// @param _uri The token URI of the created token.
+    /// @return A boolean that indicates if the operation was successful.
+    function create(
+        uint256 _id,
+        uint256 _initialSupply,
+        string calldata _uri
+    ) external returns (bool);
+
+    /// @notice Batch mint tokens. Assign directly to _to[].
+    /// @dev Throws if `msg.sender` is not allowed to mint
+    /// @param _id The token id to mint.
+    /// @param _to The list of addresses that will receive the minted tokens.
+    /// @param _quantities The list of quantities of tokens being minted.
+    function mint(
+        uint256 _id,
+        address[] calldata _to,
+        uint256[] calldata _quantities
+    ) external;
+}
+```
+
+### Burning Extension
+
+The **burning extension** is OPTIONAL for KIP-37 smart contracts. This allows your contract to burn tokens.
+
+The optional `KIP37Burnable` extension can be identified with the (KIP-13 Standard Interface Detection)[https://kips.klaytn.com/KIPs/kip-13].
+
+If the optional `KIP37Burnable` extension is included:
+
+- The KIP-13 `KIP37Burnable` function MUST return the constant value `true` if `0x9e094e9e` is passed through the `interfaceID` argument.
+- When minting/creating tokens, the `_from` argument MUST be set to `0x0` (i.e. zero address). See [Minting-creating and burning-destroying rules](#minting-creating-and-burning-destroying-rules).
+
+```solidity
+pragma solidity 0.5.6;
+
+/// @title KIP-37 Multi Token Standard, optional burning extension
+///  Note: the KIP-13 identifier for this interface is 0x9e094e9e.
+interface IKIP37Burnable {
+    /// @notice Burns specific KIP37 tokens.
+    /// @dev Throws if `msg.sender` is not allowed to burn the token
```

Comment:
```suggestion
    /// @dev Throws if `msg.sender` is not allowed to burn the token.
    ///   Throws if `_account` does not have `_value` tokens for `_id`.
```

Reply:
Do we need the parameter `_account`?
If needed, it seems that we need two functions `burn` and `burnFrom`.